### PR TITLE
Rspec upgrade

### PR DESCRIPTION
**Pull request by jvanbaarsen**
**What does this PR do?**
It upgrades rspec to version 3.x

**Why is this PR needed?**
Well, its not needed but its nice to be on the latest version.

**Extra notes**
We will be ignoring the Hound warnings in this PR since it touches a lot of old code and it is just not doable to fix it all. Focus is on getting the tests to pass and upgrading to rspec 3.

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://jvanbaarsen-rspec-upgrade.c68bfdc4cac4c32c41e1a922e73062ea.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://jvanbaarsen-rspec-upgrade.c68bfdc4cac4c32c41e1a922e73062ea.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: spec/features/issues_spec.rb [hunk 1]
```diff
@@ -25,30 +25,34 @@
...
       before do
-        fill_in "issue_title", with: "bug 345"
-        fill_in "issue_description", with: "bug description"
+        fill_in 'issue_title', with: 'bug 345'
+        fill_in 'issue_description', with: 'bug description'
       end
 
-      it { expect { click_button "Save changes" }.to_not change {Issue.count} }
+      it 'does not change issue count' do
+        expect {
```
Comment (jvanbaarsen) [inline spec/features/issues_spec.rb]: Going to ignore this one. Not sure about a better fix

File: spec/helpers/diff_helper_spec.rb [hunk 2]
```diff
@@ -10,58 +10,61 @@
...
-      parallel_diff(diff_file, 0).should match_array(parallel_diff_result_array)
+      expect(parallel_diff(diff_file, 0)).
+        to match_array(parallel_diff_result_array)
     end
   end
 
   describe 'generate_line_code' do
     it 'should generate correct line code' do
-      generate_line_code(diff_file.file_path, diff_file.diff_lines.first).should == '2f6fcd96b88b36ce98c38da085c795a27d92a3dd_6_6'
+      expect(generate_line_code(diff_file.file_path, diff_file.diff_lines.first)).
```
Comment (jvanbaarsen) [inline spec/helpers/diff_helper_spec.rb]: Going to ignore this one, not sure how to make it any shorter.

**Comment by jvanbaarsen** [comment]
@randx I think it would be great if this can be merged in already. That way we are at rspec 2.99 the version you need to be on to fix the issues you're going to hit when upgrading to 3.x. I'll keep working on the upgrade to version 3.x but Im running into some issues that I wont't be able to fix tonight. What do you think?

**Comment by randx** [comment]
@jvanbaarsen merged!

**Comment by jvanbaarsen** [comment]
Thanks!

**Pull request merged**